### PR TITLE
[11.x] Improve authenticateViaBearerToken() performance

### PR DESCRIPTION
**Pull request by alecpl**
One select from oauth_access_tokens and one from oauth_clients less. The issues mentioned in #382.

Before:
```
select * from `oauth_access_tokens` where `id` = ? limit 1
select * from `oauth_access_tokens` where `id` = ? limit 1
select * from `oauth_clients` where `id` = ? limit 1
select * from `oauth_access_tokens` where `id` = ? limit 1
select * from `oauth_clients` where `id` = ? limit 1
```
After:
```
select * from `oauth_access_tokens` where `id` = ? limit 1
select * from `oauth_clients` where `id` = ? limit 1
select * from `oauth_access_tokens` where `id` = ? limit 1
```
as you see there's still one redundant query, but it might be harder to get rid of it.

### CODE REVIEW

**Comment by driesvints** [comment]
@billriess I think it's best if you go over these changes if you can spare the time?

File: src/Guards/TokenGuard.php [hunk 1]
```diff
@@ -170,15 +163,6 @@ protected function authenticateViaBearerToken($request)
             $psr->getAttribute('oauth_access_token_id')
         );
 
-        $clientId = $psr->getAttribute('oauth_client_id');
-
-        // Finally, we will verify if the client that issued this token is still valid and
-        // its tokens may still be used. If not, we will bail out since we don't want a
-        // user to be able to send access tokens for deleted or revoked applications.
-        if ($this->clients->revoked($clientId)) {
```
Comment (billriess) [inline src/Guards/TokenGuard.php:178]: Is this check still happening somewhere else?

Comment (alecpl) [reply src/Guards/TokenGuard.php:178]: $this->clients->findActive() above makes sure the $client is not revoked.

**Comment by billriess** [comment]
I can take a look at this in the next day or two but glancing over it seems good.

**Comment by taylorotwell** [comment]
I see this was sent to `master`. Is there a breaking change? Should this go to `10.x`?

**Comment by alecpl** [comment]
@driesvints didn't like the protected method removal.

**Comment by driesvints** [comment]
@alecpl @taylorotwell technically we could do the changes on 10.x if we left the method in there. 

@billriess thanks for reviewing 👍

**Pull request merged**